sound/soc/sof/ipc4-topology.c

```diff
@@ -834,26 +835,52 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 	if (ret)
 		goto err;
 
-	cfg_size = sizeof(struct sof_ipc4_base_module_cfg);
+	ret = sof_ipc4_widget_setup_msg(swidget, &process->msg);
+	if (ret)
+		goto err;
+
+	/* parse process init module payload config type from module info */
+	fw_module = swidget->module_info;
+	process->init_config = FIELD_GET(SOF_IPC4_MODULE_INIT_CONFIG_MASK,
+					 fw_module->man4_module_entry.type);
```

Comment:
@ujfalusi how will this work for modules that absolutely need the output format for pin 0? For example the mic selector. this will need to come from the topology and cannot possibly be skipped right?

Reply:
I see, true. Like SRC for example?